File: ruby/arcade/digitsProduct.rb
def digitsProduct(product)
    return 10 if product == 0
    return 1 if product == 1
    factor, res, work = "", 0, product
    9.downto 2 do |d|
        while work % d == 0 do
            work /= d
            factor.insert(0, d.to_s)
        end
    end
    res = (factor.to_i < product) ? -1 : factor.to_i if factor.size > 0
    res = -1 if factor.size == 0
    return res
end

File: java/arcade/DigitsProduct.java
public final class DigitsProduct {

	int digitsProduct(int product) {
	    if(product == 0) { return 10; }
	    else if(product == 1) { return 1; }
	    StringBuilder fct = new StringBuilder("");
	    int res = 0, work = product;
	    for(int i = 9; i > 1; i--) {
	        while(work % i == 0) {
	            work /= i;
	            fct.insert(0, i+"");
	        }
	    }
	    if(fct.length() > 0) {
	        res = Integer.valueOf(fct.toString());
	        if(res < product) { res = -1; }
	    }
	    else { res = -1; }
	    return res;
	}
}